distributed/src/test/java/com/gentics/mesh/distributed/PluginClusterTest.java

```diff
@@ -0,0 +1,131 @@
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+/**
+ * These tests require the test plugins to be build. You can build these plugins using the /core/build-test-plugins.sh script.
+ */
+public class PluginClusterTest extends AbstractClusterTest {
+
+	private static String clusterPostFix = randomUUID();
+
+	private static final int STARTUP_TIMEOUT = 500;
+
+	private static final Logger log = LoggerFactory.getLogger(PluginClusterTest.class);
+
+	@ClassRule
+	public static MeshDockerServer serverA = new MeshDockerServer(vertx)
+		.withClusterName(clusterPostFix)
+		.withNodeName("nodeA")
+		.withDataPathPostfix(randomToken())
+		.withInitCluster()
+		.waitForStartup()
+		.withWriteQuorum(2)
+		.withPlugin(new File("../core/target/test-plugins/basic/target/basic-plugin-0.0.1-SNAPSHOT.jar"), "basic.jar")
+		.withClearFolders();
+
+	@BeforeClass
+	public static void waitForNodes() throws InterruptedException {
+		LoggingConfigurator.init();
+		serverA.awaitStartup(STARTUP_TIMEOUT);
+	}
+
+	@Before
+	public void setupLogin() {
+		login(serverA);
+	}
+
+	@Test
+	public void testPluginDeployment() throws InterruptedException {
+		// With one node the quorum is not reached and thus the plugin should not be registered.
+		assertNoPluginRegistration(serverA, 3000);
+		MeshDockerServer serverB = addSlave("nodeB");
+		try {
+			waitForPluginRegistration(serverA, 3000);
+			waitForPluginRegistration(serverB, 3000);
+		} finally {
+			serverB.stop();
+		}
+	}
+
+	/**
+	 * Assert that no plugin registration happens within the given time.
+	 * 
+	 * @param server
+	 * @param timeInMilliseconds
+	 * @throws InterruptedException
+	 */
+	private void assertNoPluginRegistration(MeshDockerServer container, int timeInMilliseconds) throws InterruptedException {
+		Thread.sleep(timeInMilliseconds);
+		PluginListResponse plugins = call(() -> container.client().findPlugins());
+		assertEquals("One plugin should be listed.", 1, plugins.getData().size());
+		assertEquals("The plugin should still be registered.", PRE_REGISTERED, plugins.getData().get(0).getStatus());
+	}
+
+	private MeshDockerServer addSlave(String nodeName) throws InterruptedException {
+		MeshDockerServer server = prepareSlave(clusterPostFix, nodeName, randomToken(), true, 2)
+			.withPlugin(new File("../core/target/test-plugins/basic/target/basic-plugin-0.0.1-SNAPSHOT.jar"), "basic.jar");
+		server.start();
+		server.awaitStartup(STARTUP_TIMEOUT);
+		login(server);
+		return server;
+	}
+
+	/**
+	 * Check whether the plugin can be found and is registered. Otherwise fail after the given timeout is exceeded.
+	 * 
+	 * @param container
+	 * @param timeoutInMilliseconds
+	 * @throws InterruptedException
+	 */
+	private void waitForPluginRegistration(MeshDockerServer container, int timeoutInMilliseconds) throws InterruptedException {
+		long start = System.currentTimeMillis();
+		PluginResponse plugin = null;
+		while (true) {
```

Comment:
Not possible since this is not embedded. Using websocket would be error prone since the docker container is started async this way we can't 100% ensure that we catch the event of the started container.